### ISSUE TITLE
Audit trail [ACAD # 19]

### ISSUE THREAD

**Issue by craigpatten** [acadsamplesdb/acad_samples#3]
Track edits:
- who
- what
- when

**Comment by sgoodswen** [comment]
The level required for logging changes depends on the answers to the following questions:
1: How often is the database backed up?
2. How long are the backups held in archive?

**Comment by Li-ReDBox** [comment]
Recording actions can be done. How are they going to be used? In Admin interface (/admin/), on the right hand, you can see such information. My understanding is it is for read-only.

Some discussions may be helpful:
http://stackoverflow.com/questions/987669/tying-in-to-django-admins-model-history

**Comment by sgoodswen** [comment]
My understanding is that whenever an delete, insert or update is performed on the postgreSQL database,  the user, date, time, and the action performed are recorded in a log file.  At this stage, the log file will only be manually viewed (e.g. using a text editor) by an administrator, if required. Its main purpose is to manually determine the relevant backup to restore following a catastrophic change.

I don’t think it is necessary that we view the log file via the Admin interface (/admin/) because I expect this file eventually to be huge.

**Comment by Li-ReDBox** [comment]
Do you need to log data or just action?

**Comment by sgoodswen** [comment]
I don't think we need to log all the data but logging the ID (or primary key) along with the action would be useful
